airbyte-integrations/connectors/source-datadog/unit_tests/test_streams.py

```diff
@@ -98,3 +110,96 @@ def test_next_page_token_paginated(stream, config):
+def test_site_parameter_is_set(config):
+    site = "example.com"
+    stream = DatadogStream(
+        site=site,
+        query=config["query"],
+        max_records_per_request=config["max_records_per_request"],
+        start_date=config["start_date"],
+        end_date=config["end_date"],
+        query_start_date=config["start_date"],
+        query_end_date=config["end_date"],
+    )
+    url_base = stream.url_base
+    expected_url_base = f"https://api.{site}/api"
+    assert url_base == expected_url_base
+
+
+@patch.multiple(DatadogStream, __abstractmethods__=set())
+def test_site_parameter_is_not_set(config):
+    stream = DatadogStream(
+        site=config["site"],
+        query=config["query"],
+        max_records_per_request=config["max_records_per_request"],
+        start_date=config["start_date"],
+        end_date=config["end_date"],
+        query_start_date=config["start_date"],
+        query_end_date=config["end_date"],
+    )
+    url_base = stream.url_base
+    expected_url_base = "https://api.datadoghq.com/api"
+    assert url_base == expected_url_base
+
+
+@patch.multiple(SeriesStream, __abstractmethods__=set())
+def test_request_body_json(config):
+    stream = SeriesStream(
+        site=config["site"],
+        query=config["query"],
+        max_records_per_request=config["max_records_per_request"],
+        start_date=config["start_date"],
+        end_date=config["end_date"],
+        query_start_date="2023-01-01T00:00:00Z",
+        query_end_date="2023-02-01T00:00:00Z",
+        name="test_stream",
+        data_source="metrics",
+        query_string="test_query"
+    )
+    stream_state = {
+        "stream_state_key": "value"
+    }
+    expected_payload = {
+        "data": {
+            "type": "timeseries_request",
+            "attributes": {
+                "to": 1675209600000,
+                "from": 1672531200000,
+                "queries": [
+                    {
+                        "data_source": "metrics",
+                        "query": "test_query",
+                        "name": "test_stream"
+                    }
+                ]
+            }
+        }
+    }
+    payload = stream.request_body_json(stream_state)
+    assert payload == expected_payload
+
+
+@patch.multiple(SeriesStream, __abstractmethods__=set())
+def test_get_json_schema(config):
+    stream = SeriesStream(
+        site=config["site"],
+        query=config["query"],
+        max_records_per_request=config["max_records_per_request"],
+        start_date=config["start_date"],
+        end_date=config["end_date"],
+        query_start_date=config["start_date"],
+        query_end_date=config["end_date"],
+        name="test_stream",
+        data_source="metrics",
+        query_string="test_query"
+    )
+    expected_schema = {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "type": "object",
+        "properties": {},
+        "additionalProperties": True
+    }
```

Comment:
Can you improve this test to a better example of what type of properties you can expect?

Reply:
Do we actually need a specific schema for every response ? I saw generic schemas being used in datadog [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-datadog/source_datadog/schemas/metrics.json)